service/query/blocks.js

```diff
@@ -31,9 +31,11 @@ module.exports = function(app, client) {
 		}).then(async (response) => {
 			const blk = await client.block(round).do();
 			const proposer = algosdk.encodeAddress(blk["cert"]["prop"]["oprop"]);
+			const hashId = algosdk.encodeAddress(blk["cert"]["prop"]["dig"]);
```

Comment:
block hash shouldn't be encoded to an address.
also `hashId` is not a suiting variable name, `blockHash` fits better I think.

Reply:
Would you mind telling me how I should encode the hash?

I tried using all the algod encoding methods, but only `encodeAddress` gives no errors. I also tried using a custom method to do it, but the outputs I get back don't make sense.

Here's a testing script I made up and the corresponding outputs:

```js
const constants = require('./service/global');
const algosdk = require('algosdk');

const algoUrl = new URL(constants.algodurl);
const client = new algosdk.Algodv2(
    constants.algodapi,
    algoUrl,
    algoUrl.port ? algoUrl.port : 8080);

function Int64ToString(bytes, isSigned) {
    const isNegative = isSigned && bytes.length > 0 && bytes[0] >= 0x80;
    const digits = [];
    bytes.forEach((byte, j) => {
        if(isNegative)
            byte = 0x100 - (j == bytes.length - 1 ? 0 : 1) - byte;
        for(let i = 0; byte > 0 || i < digits.length; i++) {
            byte += (digits[i] || 0) * 0x100;
            digits[i] = byte % 10;
            byte = (byte - digits[i]) / 10;
        }
    });
    return (isNegative ? '-' : '') + digits.reverse().join('');
}

const tests = [
    {
        inp: [77, 101, 130, 33, 7, 252, 253, 82],
        signed: false,
        expectation: '5577006791947779410'
    },
    {
        inp: [255, 255, 255, 255, 255, 255, 255, 255],
        signed: true,
        expectation: '-1'
    },
];

tests.forEach(test => {
    const result = Int64ToString(test.inp, test.signed);
    console.log(`${result} ${result !== test.expectation ? '!' : ''}=== ${test.expectation}`);
});

async function test(blockNum) {
    const blk = await client.block(blockNum).do();
    const proposer = algosdk.encodeAddress(blk["cert"]["prop"]["oprop"]);

    console.log('proposer');
    console.log(proposer);

    console.log('blockHash');
    console.log(blk["cert"]["prop"]["dig"]);

    // const blockHash = algosdk.encodeObj(blk["cert"]["prop"]["dig"]);
    try {
        console.log('Using algosdk.encodeUint64');
        const blockHash = algosdk.encodeUint64(blk["cert"]["prop"]["dig"]);
        console.log('blockHash');
        console.log(blockHash);
    } catch (e) {
        console.log('Failed');
        console.log(e);
    }

    try {
        console.log('Using algosdk.encodeAddress');
        const blockHash = algosdk.encodeAddress(blk["cert"]["prop"]["dig"]);
        console.log('blockHash');
        console.log(blockHash);
    } catch (e) {
        console.log('Failed');
        console.log(e);
    }

    try {
        console.log('Using algosdk.encodeObj');
        const blockHash = algosdk.encodeObj(blk["cert"]["prop"]["dig"]);
        console.log('blockHash');
        console.log(blockHash);
    } catch (e) {
        console.log('Failed');
        console.log(e);
    }

    try {
        console.log('Using algosdk.encodeUnsignedTransaction');
        const blockHash = algosdk.encodeUnsignedTransaction(blk["cert"]["prop"]["dig"]);
        console.log('blockHash');
        console.log(blockHash);
    } catch (e) {
        console.log('Failed');
        console.log(e);
    }

    try {
        console.log('using custom function');
        const blockHash = Int64ToString(blk["cert"]["prop"]["dig"], true);
        console.log('blockhash');
        console.log(blockHash);
    } catch (e) {
        console.log('Failed');
        console.log(e);
    }

    try {
        console.log('using custom function');
        const blockHash = Int64ToString(blk["cert"]["prop"]["dig"], false);
        console.log('blockhash');
        console.log(blockHash);
    } catch (e) {
        console.log('Failed');
        console.log(e);
    }
}


test(11)
    .then(() => { return client.block(12).do(); })
    .then((res) => { console.log(res); });

```

Output:

<img width="1082" alt="Screenshot 2021-05-10 at 4 16 50 PM" src="https://user-images.githubusercontent.com/13964463/117627887-28628000-b1ab-11eb-8969-a65d3cb7e368.png">

Reply:
I just tried this:

```js
try {
    console.log('base64');
    const blockHash = Buffer.from(blk["cert"]["prop"]["dig"], 'base64').toString();
    console.log('blockhash');
    console.log(blockHash);
} catch (e) {
    console.log('Failed');
    console.log(e);
}
```

But I got some weird output:

<img width="231" alt="Screenshot 2021-05-10 at 6 00 55 PM" src="https://user-images.githubusercontent.com/13964463/117642294-b180b380-b1b9-11eb-81f2-ce9ee67e95c7.png">

I'm really sorry for having to keep bothering you on this, if there is some documentation I can read up on or code I can look at so I can understand the issue better and potentially fix it on my own it'd great too. I keep looking at the code `js-algosdk` and still can't find something I can make use of for this.

Reply:
use this: 
const blockHash = Buffer.from(blk["cert"]["prop"]["dig"]).toString("base64");